strawberryfields/result.py

```diff
@@ -75,17 +75,24 @@ class Result:
         but the return value of ``Result.state`` will be ``None``.
     """
 
-    def __init__(self, result: Mapping, ancilla_samples: Mapping = None) -> None:
+    def __init__(self, result: Mapping, **kwargs) -> None:
         self._state = None
         self._result = result
-        self._ancilla_samples = ancilla_samples
+
+        self._ancillae_samples = kwargs.get("ancillae_samples", None)
+        self._samples_dict = kwargs.get("samples_dict", None)
 
     @property
-    def raw(self) -> Mapping:
+    def raw(self) -> Mapping[str, Union[np.ndarray, List[np.ndarray]]]:
         """The raw results dictionary containing any samples and metadata.
 
+        .. warning::
+
+            The raw values are returned by reference, meaning that any changes to
+            them will be reflected in the ``Result`` object.
+
         Returns:
-            dict: samples and metadata
+            Mapping[str, Union[np.ndarray, List[np.ndarray]]: samples and metadata
```

Comment:
Just curious: is the convention in PL and SF to use `list` or `List` in the Sphinx docstrings?
```suggestion
            Mapping[str, Union[np.ndarray, list[np.ndarray]]: samples and metadata
```

Reply:
Only able to be `list` after Python 3.9, right? So dependent on the Python versions supported?

Reply:
> Only able to be list after Python 3.9, right? So dependent on the Python versions supported?

I didn't know that Python 3.9 supported nested types in `list`. That's basically why I used `List` here instead, which is probably still a good idea if we'd like to have subtypes (although this technically only affects type hints, and not types in docstrings, I think).

> is the convention in PL and SF to use list or List in the Sphinx docstrings?

I don't think there is a convention here. `list` is definitely more common, but there are quite a few `List` occurrences as well across SF and PL.

Reply:
> I don't think there is a convention here. list is definitely more common, but there are quite a few List occurrences as well across SF and PL.

Actually, this should be `list`, any instances of `List` were accidental or missed during code review! 

This is primarily because our Sphinx conventions predate Python type hinting